### PR TITLE
Generate all.sample automagically

### PR DESCRIPTION
**Pull request by stpierre**
This adds a script, generate_group_vars_sample.sh, that generates
group_vars/all.sample from roles/ceph-common/defaults/main.yml to
avoid discrepancies between the two files. It also converts the line
endings in main.yml from DOS to Unix, since generating all.sample was
spreading the line ending plague around to more files.

### CODE REVIEW

**Comment by leseb** [comment]
Would you mind doing this for all the roles? Perhaps using `$1` as role input.

**Comment by stpierre** [comment]
I've just added a loop to properly generate all roles every time.

File: group_vars/all.sample [hunk 1]
```diff
@@ -1,18 +1,33 @@
 ---
 # Variables here are applicable to all host groups NOT roles
```
Comment (leseb) [inline group_vars/all.sample]: why did you merge all the files?
IMHO `all.sample` should container `ceph-common`.

**Comment by leseb** [comment]
@stpierre thanks for this! did you generated all the `group_vars` files?

**Comment by stpierre** [comment]
Okay, I think this works like it's supposed to now. Sorry for being dense. :)

**Comment by leseb** [comment]
Because of #553 merged there is a conflict, can you rebase your work please?

**Comment by stpierre** [comment]
Rebased.

**Comment by leseb** [comment]
@stpierre not sure to understand what happened to the `defaults/main.yml` files? Why did they change?

**Comment by leseb** [comment]
Ok got it.

**Pull request merged**